src/NServiceBus.Core/Recoverability/RecoverabilityBehavior.cs

```diff
@@ -0,0 +1,93 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using Pipeline;
+
+    class RecoverabilityBehavior : Behavior<ITransportReceiveContext>
+    {
+        public RecoverabilityBehavior(FirstLevelRetriesHandler flrHandler, SecondLevelRetriesHandler slrHandler, MoveFaultsToErrorQueueHandler errorHandler,
+            bool flrEnabled, bool slrEnabled, bool runningWithTransactions)
+        {
+            this.flrHandler = flrHandler;
+            this.slrHandler = slrHandler;
+            this.errorHandler = errorHandler;
+
+            this.flrEnabled = flrEnabled;
```

Comment:
the enabled flags seem to be a bit redundant with the null checks. Would it make sense to move the null checks in here to reduce the parameters? (not a big issue though)